File: 5 Periodo/Padrões/final-project/Strategy/strategy-java/src/br/edu/ifpb/padroes/diegofrazao/modelo/Pessoa.java
package br.edu.ifpb.padroes.diegofrazao.modelo;

import br.edu.ifpb.padroes.diegofrazao.exercicios.ExercicioFavorito;

public class Pessoa {
    private String nome;
    private ExercicioFavorito exercicioFavorito;

    public Pessoa(String nome, ExercicioFavorito exercicioFavorito) {
        this.nome = nome;
        this.exercicioFavorito = exercicioFavorito;
    }

    public void exercitar() {
        System.out.print(this.nome + " decidiu: " + this.exercicioFavorito.comecar());
    }
}

File: 5 Periodo/Padrões/final-project/Strategy/strategy-typescript/src/modelo/Pessoa.ts
import {ExercicioFavorito} from "../exercicios/ExercicioFavorito"

export class Pessoa {
  public nome: String;
  public exercicioFavorito: ExercicioFavorito;

  constructor(nome: string, exercicioFavorito: ExercicioFavorito) {
    this.nome = nome;
    this.exercicioFavorito = exercicioFavorito;
  }

  exercitar(): void {
    console.log(`${ this.nome } decidiu: `);
    this.exercicioFavorito.comecar();
  }

}